pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -113,14 +122,17 @@ type TemplateInstantiationSpec struct {
 
 	// The Kind of the template to be used, possible values are BuildTemplate
 	// or ClusterBuildTemplate. If nothing is specified, the default if is BuildTemplate
+	// +optional
 	Kind TemplateKind `json:"kind,omitempty"`
 
 	// Arguments, if specified, lists values that should be applied to the
 	// parameters specified by the template.
+	// +optional
```

Comment:
Can we make adding `+optional` a separate later PR?

Does anything enforce that non-`+optional` fields are specified? Or is it just for documentation purposes?

Reply:
I am okay with separate PR for `+optional`. I will add `optional` tags for fields introduced in this PR only.

> Does anything enforce that non-+optional fields are specified? 

k8s api should be enforcing them in future. 

Ref: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#optional-vs-required

Reply:
https://github.com/knative/build/pull/453